google-home-community.groovy

```diff
@@ -1904,9 +2157,57 @@ private executeCommand_ActivateScene(deviceInfo, command) {
     return [[:], [:]]
 }
 
+@SuppressWarnings('UnusedPrivateMethod')
+private executeCommand_ArmDisarm(deviceInfo, command) {
+    def armDisarmTrait = deviceInfo.deviceType.traits.ArmDisarm
+    def checkValue = "${armDisarmTrait.armValues["disarmed"]}"
+    def codePosition
+
+    // if the user canceled arming, issue the cancel command
+    if (command.params.cancel == true) {
+        codePosition = checkMfa(deviceInfo, "Cancel", command)
+        if (armDisarmTrait.returnUserIndexToDevice) {
+            deviceInfo.device."${armDisarmTrait.cancelCommand}"(codePosition)
+        } else {
+            deviceInfo.device."${armDisarmTrait.cancelCommand}"()
+        }
+    } else {
+        // if Google returns arm=false, that indicates disarm, otherwise execute the matching alarm level command
+        if (command.params.arm == false) {
```

Comment:
Does this need to be nested inside the `else` instead of an `if ... else if ... else`?

Reply:
The arm cancel trumps the arm commands (they are separate).  The arm command is disarm, level 0, level 1....

If cancel was part of the arm, yes, I'm just not seeing how I can structure it without the nest if else.

Reply:
Aren't they logically equivilent?  `if ... else { if ... else ... }` should have the same truth table as `if ... else if ... else ...`:

| cancel | arm | nested | not nested
| ------- |------ | -------- | ----------
| true     | true | if         | if
| true     | false | if        | if
| false    | true  | else { if } | else if
| false    | false | else { else } | else

Reply:
In fact, I would probably structure this whole thing as
```
if (command.params.cancel) {
    // cancel logic
} else if (command.params.arm) {
    // arm logic
} else {
    // disarm logic
}
```

Reply:
Agreed, that's cleaner.  Done.